File: src/ts/Models/Entities/Artist.ts
import * as _ from 'lodash';

export default class Artist {
    private _id: string;
    private _name: string;
    private _uri: string;

    public get Id(): string {
        return this._id;
    }
    public get Name(): string {
        return this._name;
    }
    public get Uri(): string {
        return this._uri;
    }

    public constructor(name: string, uri: string) {
        if (!name || name === '')
            throw new Error('name is required.');
        if (!uri || uri === '')
            throw new Error('uri is required.');

        this._id = _.uniqueId('artist_');
        this._name = name;
        this._uri = uri;
    }
}

File: src/aspCore/Models/Artists/Artist.cs
using MopidyFinder.Models.Bases;
using MopidyFinder.Models.Relations;
using Newtonsoft.Json;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MopidyFinder.Models.Artists
{
    [Table("artists")]
    [JsonObject(MemberSerialization.OptIn)]
    public class Artist : IEntity
    {
        [Key]
        [JsonProperty("Id")]
        public int Id { get; set; }

        [Required]
        [JsonProperty("Name")]
        public string Name { get; set; }

        [Required]
        [JsonProperty("LowerName")]
        public string LowerName { get; set; }

        [Required]
        [JsonProperty("Uri")]
        public string Uri { get; set; }

        [JsonProperty("ImageUri")]
        public string ImageUri { get; set; }

        [JsonProperty("ArtistAlbums")]
        public List<ArtistAlbum> ArtistAlbums { get; set; }

        [JsonProperty("GenreArtists")]
        public List<GenreArtist> GenreArtists { get; set; }
    }
}
